src/html5/parser.rs

```diff
@@ -444,6 +444,11 @@ impl<'chars> Html5Parser<'chars> {
         let mut handle_as_script_endtag = false;
 
         match &self.current_token.clone() {
+            Token::Text(value) if self.current_token.is_mixed() => {
```

Comment:
This is now a common site when dealing with text.

At these points, special cases are needed for whitespace tokens, null tokens, or both. This section will check if there are mixed characters in the current token, if so, it will split these and insert them in the token queue. From that point on, the parser continues, and on the next loop, we are back to this point again. However, now the is_mixed() will return false (since the tokens have been separated), and we continue with the following match arms.